cms/djangoapps/contentstore/utils.py

```diff
@@ -283,6 +283,23 @@ def reverse_usage_url(handler_name, usage_key, kwargs=None):
     return reverse_url(handler_name, 'usage_key_string', usage_key, kwargs)
 
 
+def get_group_display_name(user_partitions, xblock_display_name):
```

Comment:
@mushtaqak @muzaffaryousaf This is causing a problem for me on a PR I'm working on, and it would appear to be a general bug due to the implementation below. group['id'] is an integer; it can be any integer. xblock_display_name can contain any text. It is quite possible that a xblock's display name contains the integer of a group['id'], without them being related in any way.

As a concrete example, I am adding a new UserPartition based on enrollment track. The group IDs related to this partition will be small number (1, 2, 3, etc.). It's quite possible that an xblock's display name will contain those integers, and in fact, I hit failing tests that pointed out this issue (thank goodness for tests!).

Can you guys look into this tomorrow? I won't be able to merge my PR (#14682) until this is changed. For tracking purposes, I have created